autowarefoundation/autoware
# Feature/autoware msgs refactoring

## Status
DEVELOPMENT

## Description
Created a tool to automatize the process of `autoware_msgs` refactoring. Also fixed several missing dependencies and spelling errors of Autoware messages on the source code.

## Related PRs
None

## Todos


## Steps to Test or Reproduce
Usage of the refactoring script is described on its document `concerning-autoware-msgs.md`

1. Get a fresh copy of Autoware `develop` branch, ex. `git clone https://github.com/CPFL/Autoware.git -b develop --recurse-submodules [AUTOWARE_PATH]`.
2. Make sure the original builds correctly: run `./catkin_make_release` (follow the steps in [How to build](https://github.com/CPFL/Autoware/wiki/Source-Build) ).
3. Switch to `feature/autoware_msgs_refactoring` branch, ex. `git checkout feature/autoware_msgs_refactoring` and pull for changes `git pull`.
3. Run the autoware messages refactoring script as:
```
./fix-autoware-msgs.py --fix
```
from the `[AUTOWARE_PATH]/ros` folder (it assumes `src` is a subdirectory).
If only a single message namespace is to be refactored, use the `--namespace` option as:
```
./fix-autoware-msgs.py --fix --namespace N
```
where `N` is a string in [`autoware_detection_msgs`, `autoware_localization_msgs`, `autoware_planning_msgs`, `autoware_sensing_msgs`, `autoware_sync_msgs`, `autoware_socket_msgs`]
4. Make sure refactored version builds correctly: run again `./catkin_make_release`.
5. (Optionally) If necessary, everything can be reverted to its original unaltered state with `./fix-autoware-msgs.py --recovery`. The recovery file `.autoware_msgs_refactoring_recovery` has a list of modified files, it can be deleted.
